core/templates/google-analytics.initializer.ts

```diff
@@ -18,7 +18,13 @@
 
 import constants from 'assets/constants';
 
-(function() {
+export function initializeGoogleAnalytics() {
+  if (!constants.CAN_SEND_ANALYTICS_EVENTS) {
+    // Mock gtag function will prevent sending analytics to google.
+    window.gtag = function() {}
+    return;
+  }
```

Comment:
This looks good. One question though. What if `constants.ANALYTICS_ID` or `constants.SITE_NAME_FOR_ANALYTICS` are wrong? What happens then? When this file was not loaded it lead to disabled login, will it also lead to it? Or will it have some other effect?

Reply:
No, gtag library won't tell the user that id and site name is incorrect nor it will throw any error.

The analytics just won't appear on our dashboard, in worse case, it may register to some other dashboard (if wrong id is geniune id for other user).

Reply:
Ok, thanks!